File: src/server/models/agent_event.py
from pydantic import BaseModel, Field
from datetime import datetime, timezone
from enum import Enum
from typing import Optional, Any


class AgentEventType(str, Enum):
    POST_TOOL_USE = "PostToolUse"
    STOP = "Stop"


class AgentEvent(BaseModel):
    """Event received from Claude Code hooks."""

    event_type: AgentEventType
    session_id: str
    agent_id: Optional[str] = None  # Human-readable agent name from CMUX_AGENT_NAME
    tool_name: Optional[str] = None
    tool_input: Optional[Any] = None
    tool_output: Optional[Any] = None
    transcript_path: Optional[str] = None
    timestamp: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class AgentEventResponse(BaseModel):
    """Response after receiving an agent event."""

    success: bool
    event_id: str
    message: str = "Event received"

File: src/frontend/src/types/agent_event.ts
export type AgentEventType = 'PostToolUse' | 'Stop';

export interface AgentEvent {
  id: string;
  event_type: AgentEventType;
  session_id: string;
  tool_name?: string;
  tool_input?: unknown;
  tool_output?: unknown;
  timestamp: string;
}

export interface AgentEventsResponse {
  events: AgentEvent[];
  total: number;
}

export interface AgentSession {
  session_id: string;
  event_count: number;
  last_event: string;
  last_event_type: AgentEventType;
}

export interface AgentSessionsResponse {
  sessions: AgentSession[];
}
